### ISSUE TITLE
Issues with the new esm support

### ISSUE THREAD

**Issue by meza** [remix-run/v1-compat-utils#24]
I'm in the process of upgrading my project to ESM but am running into this issue:

`Cannot find module 'node_modules\@remix-run\dev\dist\config\routes' imported from node_modules\@remix-run\v1-route-convention\dist\index.mjs`

In my `remix.config.js` I import it like this: `import { createRoutesFromFolders } from '@remix-run/v1-route-convention';`

It worked fine in cjs land but now I'm stuck.

```bash
Error: Error loading Remix config at remix.config.js
Error [ERR_MODULE_NOT_FOUND]: Cannot find module 'node_modules\@remix-run\dev\dist\config\routes' imported from node_modules\@remix-run\v1-route-convention\dist\index.mjs
Did you mean to import @remix-run/dev/dist/config/routes.js?
    at Object.readConfig (node_modules\@remix-run\dev\dist\config.js:76:13)
    at async Object.watch (node_modules\@remix-run\dev\dist\cli\commands.js:216:78)
    at async Object.run (node_modules\@remix-run\dev\dist\cli\run.js:498:7)
```

**Comment by brophdawg11** [comment]
This may just be a missing extension on the import.  

@meza Do you want to try making a small change to verify?  Try adding the extension to the import in the `node_modules\@remix-run\v1-route-convention\dist\index.mjs` file:

```diff
- import { createRouteId } from "@remix-run/dev/dist/config/routes";
+ import { createRouteId } from "@remix-run/dev/dist/config/routes.js";
```

If that fixes it we'll get a patch release to get this ESM compatible.

**Comment by meza** [comment]
Will do, give me a sec!

**Comment by meza** [comment]
@brophdawg11 

```
ReferenceError: module is not defined in ES module scope
This file is being treated as an ES module because it has a '.js' file extension and 'package.json' contains "type": "module". To treat it as a CommonJS script, rename it to use the '.cjs' file extension.
    at Object.readConfig (node_modules\@remix-run\dev\dist\config.js:76:13)
    at async Object.watch (node_modules\@remix-run\dev\dist\cli\commands.js:216:78)
    at async Object.run (node_modules\@remix-run\dev\dist\cli\run.js:498:7)
ERROR: "dev:remix" exited with 1.
```

it feels like the `@remix-run\dev` package itself needs to do a better job at esm-ifying?

**Comment by brophdawg11** [comment]
@meza Can you provide a minimal reproduction of the issue (and notably the way you've set up your Remix app to use ESM)?  I'm unsure what's triggering the latest error there after the change so a reproduction would help us dig in further.

**Comment by jenslind** [comment]
> This may just be a missing extension on the import.
> 
> @meza Do you want to try making a small change to verify? Try adding the extension to the import in the `node_modules\@remix-run\v1-route-convention\dist\index.mjs` file:
> 
> ```diff
> - import { createRouteId } from "@remix-run/dev/dist/config/routes";
> + import { createRouteId } from "@remix-run/dev/dist/config/routes.js";
> ```
> 
> If that fixes it we'll get a patch release to get this ESM compatible.

This fix worked for me.

**Comment by brophdawg11** [comment]
I'm going to close this out since that fix worked for @jenslind and we don't have a minimal reproduction.  If this can be reproduced please post a sample here and we can re-open 👍

**Comment by egorpavlikhin** [comment]
Here is the minimal reproduction with the patch file included https://github.com/egorpavlikhin/v1-compat-utils-issue-24

**Comment by egorpavlikhin** [comment]
Actually, I found a workaround. Just rename `remix.config.js` to `remix.config.cjs` and use `const { createRoutesFromFolders } = require('@remix-run/v1-route-convention');` forcing the config to be read as CommonJS.

**Comment by technotip** [comment]
^ This still doesn't work for me :-(

**Comment by brophdawg11** [comment]
Can you provide a reproduction @technotip?

**Comment by brophdawg11** [comment]
Does your `package.json` have `"type": "module"` in it?  Remix changes to ESM by default in v2, so if you are using CJS (which is totally fine!) then you probably want to set `serverModuleFormat: "cjs"` in your Remix config.

Or, you can update your remix config to use ESM (`import` + `export default {}` instead of `require` and `module.exports = {}`).

https://remix.run/docs/en/main/start/v2#servermoduleformat

**Comment by technotip** [comment]
> https://remix.run/docs/en/main/start/v2#servermoduleformat

yes the `package.json` file has `"type": "module"` in it. 
I actually used `npx create-remix@latest` to create a new project and then transferred all the project files into it.

**Comment by cliffordfajardo** [comment]
Also ran into this error:

```
💿  remix dev

Error: Error loading Remix config at /Users/cfajardo/Repos/sdi-infra-fe/sdi-infra-fe/remix.config.mjs
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/Users/cfajardo/Repos/sdi-infra-fe/sdi-infra-fe/node_modules/@remix-run/dev/dist/config/routes' imported from /Users/cfajardo/Repos/sdi-infra-fe/sdi-infra-fe/node_modules/@remix-run/v1-route-convention/dist/index.mjs
Did you mean to import @remix-run/dev/dist/config/routes.js?
    at Object.readConfig (/Users/cfajardo/Repos/sdi-infra-fe/sdi-infra-fe/node_modules/@remix-run/dev/dist/config.js:72:13)
    at async Object.dev (/Users/cfajardo/Repos/sdi-infra-fe/sdi-infra-fe/node_modules/@remix-run/dev/dist/cli/commands.js:179:18)
    at async Object.run (/Users/cfajardo/Repos/sdi-infra-fe/sdi-infra-fe/node_modules/@remix-run/dev/dist/cli/run.js:192:7)
error Command failed with exit code 1.
```

Details about my project
- using Node 20.5.0
- using 2.0.1 for all remix packages
- package.json contains `type: "module"`
- remix-confg.js has `serverModuleFormat: "esm",`

**Comment by cliffordfajardo** [comment]
Created a repo to reproduce here:
https://github.com/cliffordfajardo/remix_express_v1_to_v2_routes_bug

## Steps to Reproduce

1. Create a Remix app using V1** _(I chose the express template in my repo)_
```
npx create-remix@1.19.3
```

3. Update all `@remix-run/*` dependencies to version `2.0.1`
4. Add `@remix-run/v1-route-convention` package to continue using V1 routes
5. Update remix.config.js to use `@remix-run/v1-route-convention` package
6. Run `npm run dev` & the error will appear

```
💿  remix dev

Error: Error loading Remix config at /Users/cfajardo/Desktop/remix_express_v1_to_v2_routes_bug/remix.config.js
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/Users/cfajardo/Desktop/remix_express_v1_to_v2_routes_bug/node_modules/@remix-run/dev/dist/config/routes' imported from /Users/cfajardo/Desktop/remix_express_v1_to_v2_routes_bug/node_modules/@remix-run/v1-route-convention/dist/index.mjs
Did you mean to import @remix-run/dev/dist/config/routes.js?
    at Object.readConfig (/Users/cfajardo/Desktop/remix_express_v1_to_v2_routes_bug/node_modules/@remix-run/dev/dist/config.js:72:13)
    at async Object.dev (/Users/cfajardo/Desktop/remix_express_v1_to_v2_routes_bug/node_modules/@remix-run/dev/dist/cli/commands.js:179:18)
    at async Object.run (/Users/cfajardo/Desktop/remix_express_v1_to_v2_routes_bug/node_modules/@remix-run/dev/dist/cli/run.js:192:7)
```

NOTE: all of these steps have been applied is in the bug reproduction repo:
https://github.com/cliffordfajardo/remix_express_v1_to_v2_routes_bug

**Comment by brophdawg11** [comment]
ESM/CJS issues should be resolved now via https://github.com/remix-run/v1-compat-utils/pull/25 which is available in [@remix-run/v1-route-convention@0.1.4](https://github.com/remix-run/v1-compat-utils/releases/tag/%40remix-run%2Fv1-route-convention%400.1.4) and [@remix-run/meta@0.1.3](https://github.com/remix-run/v1-compat-utils/releases/tag/%40remix-run%2Fv1-meta%400.1.3)
